runtime/lua/vim/_defaults.lua

```diff
@@ -974,6 +974,19 @@ do
       end
     end,
   })
+
+  -- Show progress bars in supporting terminals
+  vim.api.nvim_create_autocmd('Progress', {
+    group = vim.api.nvim_create_augroup('nvim.progress', {}),
+    desc = 'Display native progress bars',
+    callback = function(ev)
+      if ev.data.status == 'running' then
+        vim.api.nvim_ui_send(string.format('\027]9;4;1%d\027\\', ev.data.percent))
```

Comment:
This will run even if a TUI is not attached. But I guess that's harmless, and wouldn't gain much if this callback manually checked if a TUI is attached.

The more "proper" way would be to only define this event handler after a TUI attaches. But I think it makes sense to always define it, since we ostensibly plan to support GUIs somehow later.